api/api-doc/system.json

```diff
@@ -179,6 +179,21 @@
                ]
             }
          ]
+      },
+      {
+         "path":"/system/dump_profile",
+         "operations":[
+            {
+               "method":"POST",
+               "summary":"Dump coverage profile (no-op if the current binary is not instrumented)",
```

Comment:
We don't actually intend that users use this API - it won't work for them, and they won't even understand what "coverage profile" refers to.
I wish we could have some sort of "internal" or "developer" REST APIs that:
1. We don't guarantee to keep between releases, and
2. Users don't see them when listing the APIs.
@amnonh what do you think, and what's the best way to achieve that?

Reply:
I think there is a way, I will try to see if I can get it to work.

Reply:
In a follow up probably. Right now we already have error injection in this api-doc.
But I think this deserves a follow up.

Reply:
So you mean tha we need to have some dev/debug only api endpoints and this requires extending the json compiler? OK, perhaps good for a follow up indeed.

Reply:
Perhaps it makes sense to describe the destination file in the help message here.

Reply:
> Perhaps it makes sense to describe the destination file in the help message here.

The destination file is not fixed, it is affected by the `cwd` and the LLVM_PROFILE_FILE environment variable.

Reply:
Added https://github.com/scylladb/scylladb/issues/16572 for the dev rest APIs separation.